File: app/src/main/java/com/kanyideveloper/covid_19tracker/RecyclerViewAdapter.java
package com.kanyideveloper.covid_19tracker;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {

    private List<StatisticsList> statisticsLists;
    Context context;

    public RecyclerViewAdapter(List<StatisticsList> statisticsLists, Context context){
        this.statisticsLists = statisticsLists;
        this.context = context;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.content_itemlist,parent,false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        final StatisticsList statisticsList = statisticsLists.get(position);
        holder.country.setText(statisticsList.getCountry());
        holder.activeCases.setText(statisticsList.getActiveCase());
        holder.recoveredCases.setText(statisticsList.getRecovered());
        holder.deaths.setText(statisticsList.getDeaths());
        holder.totalCases.setText(statisticsList.getTotalCase());
    }

    @Override
    public int getItemCount() {
        return statisticsLists.size();
    }


    public void filterList(ArrayList<StatisticsList> filteredList){
        statisticsLists = filteredList;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{


        TextView country,activeCases,recoveredCases,totalCases,deaths;
        GridLayout gridLayout;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            country = itemView.findViewById(R.id.countryTextField);
            activeCases = itemView.findViewById(R.id.activeCases);
            recoveredCases = itemView.findViewById(R.id.recoveredCases);
            totalCases = itemView.findViewById(R.id.totalCases);
            deaths = itemView.findViewById(R.id.deathCases);
            gridLayout = itemView.findViewById(R.id.grid_layout);
        }
    }
}

File: app/src/main/java/com/kanyideveloper/covid_19tracker/kotlin/RecyclerViewAdapter.kt
package com.kanyideveloper.covid_19tracker.kotlin

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target
import com.kanyideveloper.covid_19tracker.R

class RecyclerViewAdapter(private var context: Context, private var statisticsLists: ArrayList<StatisticsList>) :
        RecyclerView.Adapter<RecyclerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val viewHolder: RecyclerViewHolder?
        val layoutView = LayoutInflater.from(parent.context).inflate(R.layout.content_itemlist, parent, false)
        viewHolder = RecyclerViewHolder(layoutView, statisticsLists)
        return viewHolder
    }

    override fun getItemCount(): Int {
        return statisticsLists.size
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        holder.country.text = statisticsLists[position].country
        holder.activeCases.text = statisticsLists[position].activeCase
        holder.recoveredCases.text = statisticsLists[position].recovered
        holder.deaths.text = statisticsLists[position].deaths
        holder.totalCases.text = statisticsLists[position].totalCases
        Glide.with(context).load(statisticsLists[position].imageUrl)
                .listener(object : RequestListener<Drawable> {
                    override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
                        holder.imageViewFlag.visibility = View.GONE
                        return false
                    }

                    override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
                        return false
                    }

                }).into(holder.imageViewFlag)
    }

    fun filterList(filteredList: java.util.ArrayList<StatisticsList>) {
        statisticsLists = filteredList
        notifyDataSetChanged()
    }

}